### DISCUSSION TITLE
How to avoid Generated of SWIGTYPE_p files

### DISCUSSION THREAD

**Discussion by kothapraveen** [swig/swig#2955] [General]
Hi,
Currently we are using swigwin-3.0.9 to convert CPP files to C#

As we have predefined data types(MTInt4, MTInt2 ) and tried to suppress the General datatypes(long,short) using below lines in interface files.

%apply const std::wstring & {std::wstring &};
%apply const MTInt4 & {long &};
%apply const MTInt2 & {short int &};
%apply const MTReal8 & {double &};
%apply const MTUnsignedChar * {unsigned char *};
%apply const MTUnsignedInt4 * {unsigned long *};

But still SWIGTYPE_p_long.cs,
SWIGTYPE_p_short.cs  files are getting generated.

please help us how we can get rid off SWIGTYPE_p_long.cs, SWIGTYPE_p_short.cs files

**Comment by wsfulton** [comment]
You need to provide information in this discussion as to what exactly a `MTInt4` is. I suspect you have not provided this for SWIG either, so SWIG generates a type wrapper class. Please read the docs on them https://swig.org/Doc4.2/Java.html#Java_type_wrapper_classes which applies equally to C#. 

If MTInt4 is a long, then this is wrong:
```swig
%apply const MTInt4 & {long &};
```
It should be:
```swig
%apply const long & {MTInt4 &};
```
